### PR TITLE
Add README to R package

### PR DESCRIPTION
**Pull request by terrytangyuan**

### CODE REVIEW

File: src/interface_r/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+## h2o4gpu - R Interface to H2O4GPU
+
+This directory contains the R package for H2O4GPU - a collection of GPU solvers by H2Oai. It builds upon the easy-to-use Scikit-learn API and its well-tested CPU-based algorithms. It can be used as a drop-in replacement for scikit-learn (i.e. import h2o4gpu as sklearn) with support for GPUs on selected (and ever-growing) algorithms. H2O4GPU inherits all the existing scikit-learn algorithms and falls-back to CPU aglorithms when the GPU algorithm does not support an important existing Scikit-learn class option.
```
Comment (navdeep-G) [inline src/interface_r/README.md]: I think you can leave it at `This directory contains the R package for H2O4GPU - a collection of GPU solvers by H2Oai`. The rest is just repeated from the main README.

File: src/interface_r/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+Then install `devtools` R package via `install.packages("devtools")` if you haven't done it yet.
+
+Finally, we can install `h2o4gpu` R package via the following:
+
+``` r
+devtools::install_github("h2oai/h2o4gpu", subdir = "src/interface_r")
+```
+
+To test your installation, try the following example that builds a simple [XGBoost](https://github.com/dmlc/xgboost) random forest classifier utilizing GPU powers:
```
Comment (navdeep-G) [inline src/interface_r/README.md]: Should be: 
 ```
To test your installation, try the following example that builds a simple [XGBoost](https://github.com/dmlc/xgboost) random forest classifier:
```

File: src/interface_r/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+``` r
+require(h2o4gpu)
+
+# Setup dataset
+x <- iris[1:4]
+y <- as.integer(iris$Species) - 1
+
+# Initialize and train the classifier
+model <- h2o4gpu.random_forest_classifier() %>% fit(x, y)
```
Comment (navdeep-G) [inline src/interface_r/README.md]: Assuming this PR depends on #374 ?

Comment (terrytangyuan) [reply src/interface_r/README.md]: Yep. Thanks! Just addressed your comments.

**Pull request merged**